src/antelope/wallets/ual/MetakeepUAL.ts

```diff
@@ -0,0 +1,534 @@
+
+    /**
+     * Returns whether or not the button should show an account name input field.
+     * This is for Authenticators that do not have a concept of account names.
+     */
+    async shouldRequestAccountName() {
+        return false;
+    }
+
+    async createAccount(publicKey: string): Promise<string> {
+        const suggestedName = await this.accountNameSelector.selectAccountName();
+        return axios.post(this.accountCreateAPI, {
+            ownerKey: publicKey,
+            activeKey: publicKey,
+            jwt: this.userCredentials.jwt,
+            suggestedName: suggestedName,
+        }).then(response => response.data.accountName);
+    }
+
+    resolveAccountName() {
+        return new Promise<string>(async (resolve, reject) => {
+            let accountName = '';
+            if (!metakeep) {
+                return reject(new Error('metakeep is not initialized'));
+            }
+            if (this.userCredentials.email === '') {
+                return reject(new Error('No account email'));
+            }
+
+            // we check if we have the account name in the cache
+            const accountNames = metakeepCache.getAccountNames(this.userCredentials.email, this.chainId);
+            if (accountNames.length > 0) {
+                if (accountNames.length > 1) {
+                    // if we have more than one account, we ask the user to select one using this callback
+                    const selectedAccount = await this.accountSelector.selectAccount(accountNames);
+                    this.resetAccountSelector();
+                    metakeepCache.setSelectedAccountName(this.userCredentials.email, this.chainId, selectedAccount);
+                    return resolve(selectedAccount);
+                } else {
+                    return resolve(accountNames[0]);
+                }
+            }
+
+            // if not, we fetch all the accounts for the email
+            const credentials = await metakeep.getWallet();
+            const publicKey = credentials.wallet.eosAddress;
+
+            metakeepCache.addCredentials(this.userCredentials.email, credentials.wallet);
+
+            try {
+                // we try to get the account name from the public key
+                const response = await axios.post(`${this.rpc.endpoint}/v1/history/get_key_accounts`, {
+                    public_key: publicKey,
+                });
+                const accountExists = response?.data?.account_names.length>0;
+                let names:string[] = [];
+
+                if (accountExists) {
+                    names = response.data.account_names;
+                    names.forEach(name => metakeepCache.addAccountName(this.userCredentials.email, this.chainId, name));
+                    if (names.length > 1) {
+                        // if we have more than one account, we ask the user to select one using this callback
+                        accountName = await this.accountSelector.selectAccount(names);
```

Comment:
```ts
accountName = await this.accountSelector.selectAccount(names);
```
Again, this line allows the user to select one account to log in with. In this case, the accounts are taken from the blockchain.